compiler/semantic/expr.go

```diff
@@ -519,64 +519,82 @@ func (a *analyzer) semAssignments(assignments []ast.Assignment, summarize bool)
 	return out, nil
 }
 
-func (a *analyzer) semAssignment(assign ast.Assignment, summarize bool) (dag.Assignment, error) {
-	rhs, err := a.semExpr(assign.RHS)
+func (a *analyzer) semAssignment(e ast.Assignment) (dag.Assignment, error) {
```

Comment:
Nit: `e` suggests an expression so stick with `assign` here.
```suggestion
func (a *analyzer) semAssignment(assign ast.Assignment) (dag.Assignment, error) {
```